### PR TITLE
[SPARK-7916] [MLlib] MLlib Python doc parity check for classification and regression

### PR DESCRIPTION
**Pull request by yanboliang**
Check then make the MLlib Python classification and regression doc to be as complete as the Scala doc.

### CODE REVIEW

**Comment by yanboliang** [comment]
The length of docstring line for PEP8 standard is 72, so truncate too long line.

File: python/pyspark/mllib/regression.py [hunk 1]
```diff
@@ -193,18 +198,28 @@ def train(cls, data, iterations=100, step=1.0, miniBatchFraction=1.0,
               initialWeights=None, regParam=0.0, regType=None, intercept=False,
               validateData=True):
         """
-        Train a linear regression model on the given data.
-
-        :param data:              The training data.
-        :param iterations:        The number of iterations (default: 100).
+        Train a linear regression model with no regularization using
```
Comment (jkbradley) [inline python/pyspark/mllib/regression.py]: This should not say "with no regularization."   It looks like this is a difference between the Python & Scala APIs.

File: python/pyspark/mllib/regression.py [hunk 2]
```diff
@@ -389,7 +434,36 @@ class RidgeRegressionWithSGD(object):
     def train(cls, data, iterations=100, step=1.0, regParam=0.01,
               miniBatchFraction=1.0, initialWeights=None, intercept=False,
               validateData=True):
-        """Train a ridge regression model on the given data."""
+        """
+        Train a regression model with L2-regularization using
+        Stochastic Gradient Descent.
+        This solves the l1-regularized least squares regression
```
Comment (jkbradley) [inline python/pyspark/mllib/regression.py]: l2, not l1
Can you please fix this bug in the Scala/Java doc too?

**Comment by jkbradley** [comment]
@yanboliang Thanks for adding this doc!  With so many Python users, this should be very helpful.

**Comment by yanboliang** [comment]
retest

**Comment by jkbradley** [comment]
This LGTM  I'll merge it into master and branch-1.4

@yanboliang Thanks for the PR, and sorry about the delay in merging!  We can update the docs on the website at any time, thankfully, so this should be able to go up once some other updates get in as well.

**Pull request closed without merging**